launcher.fnl

```diff
@@ -0,0 +1,172 @@
+                        (table.remove arg i))
+    "--indent" (do (set options.indent (table.remove arg (+ i 1)))
+                   (when (= options.indent "false")
+                     (set options.indent false))
+                   (table.remove arg i))
+    "--add-package-path" (let [entry (table.remove arg (+ i 1))]
+                           (set package.path (.. entry ";" package.path))
+                           (table.remove arg i))
+    "--add-fennel-path" (let [entry (table.remove arg (+ i 1))]
+                          (set fennel.path (.. entry ";" fennel.path))
+                          (table.remove arg i))
+    "--load" (let [file (table.remove arg (+ i 1))]
+               (dosafely fennel.dofile file options [])
+               (table.remove arg i))
+    "--no-fennelrc" (do (set options.fennelrc false)
+                        (table.remove arg i))
+    "--correlate" (do (set options.correlate true)
+                      (table.remove arg i))
+    "--check-unused-locals" (do (set options.checkUnusedLocals true)
+                                (table.remove arg i))
+    "--globals" (do (allow-globals (table.remove arg (+ i 1)))
+                    (each [global-name (pairs _G)]
+                      (table.insert options.allowedGlobals global-name))
+                    (table.remove arg i))
+    "--globals-only" (do (allow-globals (table.remove arg (+ i 1)))
+                         (table.remove arg i))
+    "--require-as-include" (do (set options.requireAsInclude true)
+                               (table.remove arg i))
+    "--metadata" (do (set options.useMetadata true)
+                     (table.remove arg i))
+    "--no-metadata" (do (set options.useMetadata false)
+                        (table.remove arg i))))
+
+(when (not options.no_searcher)
+  (let [opts []]
+    (each [k v (pairs options)]
+      (tset opts k v))
+    (table.insert (or package.loaders package.searchers)
+                  (fennel.make_searcher opts))))
+
+(fn try-readline [ok readline]
+  (when ok
+    (when readline.set_readline_name
+      (readline.set_readline_name "fennel"))
+    (readline.set_options {:keeplines 1000 :histfile ""})
+    (fn opts.readChunk [parser-state]
+      (let [prompt (if (< 0 parser-state.stackSize) ".. " ">> ")
+            str (readline.readline prompt)]
+        (if str (.. str "\n"))))
+    (var completer nil)
+    (fn opts.registerCompleter [repl-completer]
+      (set completer repl-completer))
+    (fn repl-completer [text from to]
+      (if completer
+          (do (readline.set_completion_append_character "")
+              (completer (text:sub from to)))
+          []))
+    (readline.set_complete_function repl-completer)
+    readline))
+
+;; TODO: generalize this as a plugin instead of hard-coding it
+;; we can't pcall this or we won't be able to use --require-as-include.
+(each [k v (pairs (require :fennelfriend))]
+  (tset options k v))
```

Comment:
What if we pcalled a wrapper function instead?

```suggestion
;; TODO: generalize this as a plugin instead of hard-coding it
(match (pcall #(pairs (require :fennelfriend)))
  (true step param) (each [k v (values step param)]
                      (tset options k v)))
```

Reply:
I'm on the fence about this... on the one hand, it's more resilient, but on the other hand, maybe the only reason we were using pcall in the first place is because you'd copy the old `fennel` script out of the repo and leave out the other files because we didn't have a standalone script. If we have a standalone script, what's the rationale for wanting to use just the launcher without the other goodies?

Reply:
That's a good point. The only scenario I can see it coming up is if someone does their own compilation and it somehow doesn't find fennelfriend, and ends up using the require fallback. It'd probably be better to get instant feedback on a failed `include` anyway, to catch any issues with fennelfriend.

I mostly just added it because the comment about not being able to `pcall` was still there, so just removing that comment line sounds good to me as well.